presto-main/src/main/java/com/facebook/presto/server/ConnectorMetadataUpdateHandleJsonSerde.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.presto.spi.ConnectorMetadataUpdateHandle;
+import com.facebook.presto.spi.connector.ConnectorMetadataUpdateHandleSerde;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.airlift.slice.Slices;
+
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+
+/**
+ * Json based connector handle serde
+ */
+public class ConnectorMetadataUpdateHandleJsonSerde
```

Comment:
Is it possible to make it hive specific? Some of the structure in the class is a bit weird like the loading cache

Reply:
This class would be like the default json implementation for all connectors in case thrift is not supported. Currently we have only, hive connector specific implementation of MetadataUpdateHandle.
The alternative I thought was as you mentioned, keeping it inside the connector (hive specific one in this case). However for scenarios we have multiple connector implementations, we will have to duplicate this code since the json implementation won't change. We need a common place to keep this logic and we can't keep it in spi since that would mean adding airlift to spi dependencies.
If you feel keeping it inside connector is cleaner I can go with that approach for json serde as well.